app/controllers/purchases_controller.rb

```diff
@@ -0,0 +1,50 @@
+require 'purchase/presenters/new_purchase_presenter'
+
+class PurchasesController < ApplicationController
+
+  def new
+    item = find_item(item_id)
+    purchase = new_purchase(item_id: item_id)
+    @presenter = create_presenter(purchase, item)
+  end
+
+  def create
+    purchase = new_purchase(purchase_params)
+    item = find_item(purchase_params[:item_id])
+    @presenter = create_presenter(purchase, item)
+
+    if @presenter.purchase.save
+      item.update_attributes(available: false)
```

Comment:
You have a race condition in the way purchases are made and the available attribute is used. What if two people are looking at the index page, one makes a purchase, and then the other purchases the same item?